docs/en/sql-reference/functions/tuple-map-functions.md

```diff
@@ -81,37 +59,50 @@ Alias: `MAP_FROM_ARRAYS(keys, values)`
 
 **Arguments**
 
-- `keys` — Given key array to create a map from. The nested type of array must be: [String](../data-types/string.md), [Integer](../data-types/int-uint.md), [LowCardinality](../data-types/lowcardinality.md), [FixedString](../data-types/fixedstring.md), [UUID](../data-types/uuid.md), [Date](../data-types/date.md), [DateTime](../data-types/datetime.md), [Date32](../data-types/date32.md), [Enum](../data-types/enum.md)
-- `values`  - Given value array or map to create a map from.
+- `keys` —  Array of keys to create the map from. [Array(T)](../data-types/array.md) where `T` can be any type supported by [Map](../data-types/map.md) as key type.
+- `values`  - Array or map of values to create the map from. [Array](../data-types/array.md) or [Map](../data-types/map.md).
 
 **Returned value**
 
-- A map whose keys and values are constructed from the key array and value array/map.
+- A map with keys and values constructed from the key array and value array/map.
 
 **Example**
 
 Query:
 
 ```sql
 select mapFromArrays(['a', 'b', 'c'], [1, 2, 3])
+```
 
+Result:
 
+```
 ┌─mapFromArrays(['a', 'b', 'c'], [1, 2, 3])─┐
 │ {'a':1,'b':2,'c':3}                       │
 └───────────────────────────────────────────┘
+```
 
+Query
+
+```sql
 SELECT mapFromArrays([1, 2, 3], map('a', 1, 'b', 2, 'c', 3))
+```
 
+Result:
+
+```
 ┌─mapFromArrays([1, 2, 3], map('a', 1, 'b', 2, 'c', 3))─┐
 │ {1:('a',1),2:('b',2),3:('c',3)}                       │
 └───────────────────────────────────────────────────────┘
 ```
 
 ## extractKeyValuePairs
```

Comment:
This function is quite tricky as it does not work with string representation for a map.
If you will take from example 

```
SELECT
    '{\'name\':\'neymar\',\'age\':\'31\',\'team\':\'psg\',\'nationality\':\'brazil\'}' AS str,
    extractKeyValuePairs(str) AS extracted,
    toString(extracted),
    extracted.keys,
    extracted.values

Query id: 677133ac-0ac5-489c-b1e1-4b35ba0e8574

Row 1:
──────
str:                 {'name':'neymar','age':'31','team':'psg','nationality':'brazil'}
extracted:           {'{\'name\'':'\'neymar\'','\'age\'':'\'31\'','\'team\'':'\'psg\'','\'nationality\'':'\'brazil\'}'}
toString(extracted): {'{\'name\'':'\'neymar\'','\'age\'':'\'31\'','\'team\'':'\'psg\'','\'nationality\'':'\'brazil\'}'}
extracted.keys:      ['{\'name\'','\'age\'','\'team\'','\'nationality\'']
extracted.values:    ['\'neymar\'','\'31\'','\'psg\'','\'brazil\'}']
```

Here is working way

```
SELECT
    '{\'name\':\'neymar\',\'age\':\'31\',\'team\':\'psg\',\'nationality\':\'brazil\'}' AS str,
    extractKeyValuePairs(str, ':', ',', '\'') AS extracted,
    toString(extracted),
    extracted.keys,
    extracted.values

Query id: 2e7609a9-3808-4735-aeaa-a4ff965e8b8f

Row 1:
──────
str:                 {'name':'neymar','age':'31','team':'psg','nationality':'brazil'}
extracted:           {'name':'neymar','age':'31','team':'psg','nationality':'brazil'}
toString(extracted): {'name':'neymar','age':'31','team':'psg','nationality':'brazil'}
extracted.keys:      ['name','age','team','nationality']
extracted.values:    ['neymar','31','psg','brazil']
```

Reply:
I guess `extractKeyValuePairs` was never supposed to be the opposite of `toString(map)`, otherwise it would use other defaults for the kv-delimiter, pair-delimiter and quoting character.

Reply:
We just do not have this fromString syntax. I always use this function to convert it back and would add a note how to do so. If you have other way to parse result of toString, we can document it as well.

Reply:
Okay, added.